cmd/deploy/deploy.go

```diff
@@ -348,3 +373,144 @@ func getProxyHandler(name, token string, clusterConfig *rest.Config) (http.Handl
 	return handler, nil
 
 }
+
+func (*deployCommand) waitUntilDeployed(ctx context.Context, opts *Options) error {
+	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, opts.Name)
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(opts.Timeout)
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("application '%s' didn't finish after %s", opts.Name, opts.Timeout.String())
+		case <-ticker.C:
+			isReady := true
+			dList, err := deployments.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, dep := range dList {
+				if dep.Status.ReadyReplicas != dep.Status.Replicas {
```

Comment:
The right check here is:
`observedGeneration` >= `generation` and `replicas` == `status.updatedReplicas` == `status.availableReplicas`
More information here:
https://stackoverflow.com/questions/37448357/ensure-kubernetes-deployment-has-completed-and-all-pods-are-updated-and-availabl
Or you can also check the implementation of `kubectl rollout restart --wait`

Reply:
Also, we should detect if there are errors and exit our command with an error